Base/usr/share/shell/completion/builtin.sh

```diff
@@ -74,3 +74,17 @@ _complete_kill() {
     __complete_job_spec $*[-1]
   }
 }
+
+_complete_cd() {
+    if test $*[-1] = '--' {
+        invariant_offset=0
+        dir_glob=(.*/ */)
+    } else {
+        invariant_offset=${length "$*[-1]"}
+        dir_glob="$*[-1]*/"
+    }
+
+    for $(glob $dir_glob) {
+        echo '{"kind":"plain","static_offset":0,"invariant_offset":'"$invariant_offset"',"completion":"'"$it"'","allow_commit_without_listing":true}'
```

Comment:
I wonder if it would make sense to remove the trailing `/` in the shown suggestions and have it be trailing trivia :thinking:

it would look something like this:
```
$ cd bar/
..   .    bar  baz  foo
```
as opposed to:
```
❯ cd bar/
../   ./    bar/  baz/  foo/
```

this should do that I think:
```suggestion
        echo '{"kind":"plain","static_offset":0,"invariant_offset":'"$invariant_offset"',"completion":"'"${remove_suffix / $it}"'","allow_commit_without_listing":true,"trailing_trivia":"/"}'
```